source/dub/commandline.d

```diff
@@ -1279,13 +1279,13 @@ class BuildCommand : GenerateCommand {
 			// the user provided a version manually
 			dep = Dependency(packageParts.version_);
 		} else {
-			if (packageParts.name.startsWith(":") ||
+			if (packageParts.name.value.startsWith(":") ||
 				dub.packageManager.getFirstPackage(packageParts.name))
 				// found locally
 				return 0;
 
 			// search for the package and filter versions for exact matches
-			auto basePackageName = getBasePackageName(packageParts.name);
+			auto basePackageName = getBasePackageName(PackageName(packageParts.name));
```

Comment:
packageParts.name should already be of PackageName type

Reply:
Resolved via b97f546a.